dpx/src/dpx_pdfobj.rs

```diff
@@ -3753,13 +3750,16 @@ pub unsafe extern "C" fn pdf_open(
         let new_version = pdf_deref_obj(pdf_lookup_dict(&mut *(*pf).catalog, "Version"));
         if !new_version.is_null() {
             let mut minor: u32 = 0;
-            if !(!new_version.is_null() && (*new_version).is_name())
-                || sscanf(
-                    pdf_name_value(&*new_version).as_ptr(),
-                    b"1.%u\x00" as *const u8 as *const i8,
-                    &mut minor as *mut u32,
-                ) != 1i32
-            {
+            if (&*new_version).is_name() {
+                let new_version_str = pdf_name_value(&*new_version).to_bytes();
+                if new_version_str.starts_with(b"1.") {
+                    minor = std::str::from_utf8(&new_version_str[2..])
+                        .unwrap_or("")
+                        .parse::<u32>()
+                        .unwrap_or(0)
+                }
+            }
+            if minor == 0 {
```

Comment:
Isn’t 0 a valid value for minor version? The original code checked parse failure, so that would require checking Err from parse().

Reply:
yup, 1.0 is valid value(though useless in practice) , and 2.0 is valid too by the way:)